clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -367,32 +367,46 @@ bool ArrayBoundsRewriter::hasNewBoundsString(PVConstraint *PV, Decl *D,
   return !BStr.empty() && !PV->hasBoundsStr();
 }
 
-std::set<PersistentSourceLoc *> RewriteConsumer::EmittedDiagnostics;
+std::set<std::tuple<std::string, int, int>> RewriteConsumer::EmittedDiagnostics;
```

Comment:
Asked a question about this earlier (this tuple type); maybe define a typedef for it and use the typedef name here?